sdk/src/ssi/credential.go

```diff
@@ -4,48 +4,61 @@ import (
 	"github.com/TBD54566975/ssi-sdk/credential"
 	"github.com/TBD54566975/ssi-sdk/credential/signing"
 	"github.com/TBD54566975/ssi-sdk/crypto"
+	"github.com/TBD54566975/ssi-sdk/did"
 	"github.com/goccy/go-json"
+	"github.com/lestrrat-go/jwx/jwk"
+	"github.com/pkg/errors"
 )
 
-// SignVerifiableCredentialJWT takes in a key ID, key type, private key, and a verifiable credential
-// The keyID and KeyType are used to reconstruct a go-friendly private key to be used for signing
+// SignVerifiableCredentialJWT takes in a key ID, private JWK, and a verifiable credential.
+// The key ID and private JWK are used to reconstruct a go-friendly private key to be used for signing
 // the credential, which will be packaged as a JWT according to the VC-JWT 1.0 specification.
-// The resulting is returned as a string representation of a JWT.
-func SignVerifiableCredentialJWT(keyID, keyType string, privateKey, vcJSONBytes []byte) string {
-	privKey, err := crypto.BytesToPrivKey(privateKey, crypto.KeyType(keyType))
+// The resulting signed credential is returned as a string representation of a JWT.
+func SignVerifiableCredentialJWT(keyID string, privateJwk []byte, vcBytes []byte) (string, error) {
+	key, err := jwk.ParseKey(privateJwk)
 	if err != nil {
-		return ""
+		return "", errors.Wrap(err, "parsing key")
 	}
-	signer, err := crypto.NewJWTSigner(keyID, privKey)
+	signer, err := crypto.NewJWTSignerFromKey(keyID, key)
 	if err != nil {
-		return ""
+		return "", errors.Wrap(err, "creating signer")
 	}
-
-	var cred credential.VerifiableCredential
-	if err = json.Unmarshal(vcJSONBytes, &cred); err != nil {
-		return ""
+	var vc credential.VerifiableCredential
+	if err := json.Unmarshal(vcBytes, &vc); err != nil {
+		return "", errors.Wrap(err, "unmarshalling vc")
 	}
-
-	signedCredential, err := signing.SignVerifiableCredentialJWT(*signer, cred)
+	signedVC, err := signing.SignVerifiableCredentialJWT(*signer, vc)
 	if err != nil {
-		return ""
+		return "", errors.Wrap(err, "signing vc")
 	}
-	return string(signedCredential)
+
+	return string(signedVC), nil
 }
 
-// VerifyVerifiableCredentialJWT takes in a key ID, key type, public key, and a JWT string
-// The keyID and KeyType are used to reconstruct a go-friendly public key to be used for verifying
-// the JWT. The JWT is then decoded and verified, and the result is returned as a boolean.
-func VerifyVerifiableCredentialJWT(keyID, keyType string, publicKey []byte, jwt string) bool {
-	pubKey, err := crypto.BytesToPubKey(publicKey, crypto.KeyType(keyType))
+// VerifyVerifiableCredentialJWT takes in a key ID, DID key, and a JWT string.
+// The key ID and DID key are used to reconstruct a go-friendly public key to be used for verifying
+// the JWT. The JWT is then decoded and verified, and the resulting signed and verified credential is returned.
+func VerifyVerifiableCredentialJWT(keyID string, didKey string, jwt string) ([]byte, error) {
+	pubKeyBytes, _, keyType, err := did.DIDKey(didKey).Decode()
```

Comment:
Similar to the above function, the clients do not have the `keyType` or the bytes of the `publicKey`. They DO have the DID key, which can be used to derive the keyType and publicKey.

I changed this function to take in the `didKey` so that it could be callable via the clients with the information they have. Again, open to other solutions!